app/src/main/java/com/duckduckgo/app/tabs/model/TabDataRepository.kt

```diff
@@ -51,6 +51,18 @@ class TabDataRepository @Inject constructor(
         return tabId
     }
 
+    override suspend fun addWithSource(url: String?, skipHome: Boolean, isDefaultTab: Boolean): String {
+        val tabId = generateTabId()
+        add(
+            tabId,
+            buildSiteData(url),
+            skipHome = skipHome,
+            isDefaultTab = isDefaultTab,
+            sourceTabId = liveSelectedTab.value?.tabId
```

Comment:
Getting the value from `liveSelectedTab` works because when we call `addWithSource` (or `deleteCurrentTabAndSelectSource`) a ViewModel is observing that LiveData externally. But if we end up in a scenario where we call this method without anyone observing the `LiveData`, this will fail. I think is better to use `tabsDao.selectedTab()`.

Reply:
Makes sense, let me know how my latest approach looks. A bit new to coroutines still, so not sure if the usage of withContext is ideal, but we'd need to await the value from the DB before proceeding and can't access the DB on the main thread.